### DISCUSSION TITLE
Understanding what the projects requires.

### DISCUSSION THREAD

**Discussion by SisiVero** [camicroscope/caMicroscope#685] [Q&A]
Hello @birm  , having read the intern task,  So, if I understand well, I am required to clone the repos for this project, then run it and see how it works. Then suggest changes to ui if any and then try and make it a standalone executable,  yes? I have few other questions regarding the draft you shared.
If understand correctly, I am to clone the repositories , the 30 of them? Then install the Docker and Docker compose and run caMicriscope locally to experiment it, yes?
Now, when you said to add a Slide to my instance of caMicriscope, what do you mean?
I'm sorry I'm asking lots of questions. I just need to be sure I'm on the right path.

**Comment by gbengaoluwadahunsi** [comment]
Hello @SisiVero 

You are not required to clone all the 30 repos to your local machine, you are to clone the Distro repo.
As regards the slide, it means you add a slide to the viewer ,play around the UI, the functionality and note your possible contributions.

**Reply by SisiVero** [reply]
Thank you so much

**Reply by PASEDA-MODUPE** [reply]
hello pls  @SisiVero  and @gbengaoluwadahunsi  after cloning the repo must i install docker for it to work

**Reply by gbengaoluwadahunsi** [reply]
hello @PASEDA-MODUPE , you do have to install docker on your local machine, you can confirm in your  IDE terminal or in the CMD by typing docker  --version after installing docker.

**Reply by PASEDA-MODUPE** [reply]
Okay noted thanks

**Reply by We13b-MD** [reply]
Thanks gbenga this reaaly helped

**Reply by gbengaoluwadahunsi** [reply]
glad  it helped.

**Reply by SisiVero** [reply]
Hello @PASEDA-MODUPE , were you able to install docker?

**Reply by NgweBecky** [reply]
@SisiVero yes I was, any issue on your side??

**Comment by We13b-MD** [comment]
In the project communication information area there was a part were  it is required by applicants to take a tutorial before before making contributions, if so how do we proceed with that.

**Reply by gbengaoluwadahunsi** [reply]
Hello  @We13b-MD, I think @birm pointed our attention to the caMicroscope documentation, you will get most of the guidelines there.

**Reply by We13b-MD** [reply]
Ok

**Reply by NgweBecky** [reply]
@We13b-MD I started by downloading the docker desktop .exe file, use this link https://docs.docker.com/desktop/install/windows-install/ for guidance. Please reach out if any issue encountered in the process :)

**Reply by blessdamy** [reply]
@NgweBecky please I have troubles getting started

**Comment by kivuvarosekivuvan** [comment]
I am facing this when trying to upload the image, could anyone possibly see my mistake?[
![Screenshot from 2024-03-06 01-06-09](https://github.com/camicroscope/caMicroscope/assets/124133577/2aef759e-0575-4881-9577-1d19de0a0703)
](url)

**Reply by We13b-MD** [reply]
hello kivuva how were you able to get to this stage how did you get docker up and running

**Reply by kivuvarosekivuvan** [reply]
These are the steps I have followed to set up docker 

- sudo apt update

- sudo apt install apt-transport-https ca-certificates curl software-properties-common

- curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg

- echo "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null

- sudo apt update

- sudo apt install docker-ce docker-ce-cli containerd.io
#### However, I've come across some errors. If you happen to encounter any issues as well, please don't hesitate to let me know, and I'll be more than happy to assist you with resolving them.

**Reply by gbengaoluwadahunsi** [reply]
@kivuvarosekivuvan , it is possible that the file type does not allow Png , Jpeg or webp, I am still trying to figure it out since  I am observing the same error.

**Reply by kivuvarosekivuvan** [reply]
Yes,even with png, jpeg and jpg does not allow

**Reply by gbengaoluwadahunsi** [reply]
Have you gone through the docs?, there is an information as regards the type of files the viewer allows

**Reply by kivuvarosekivuvan** [reply]
I initially noticed it while I was reading, but when I went back to look for it, I couldn't find it again. Do you happen to have any insights that could help me with this? I would greatly appreciate your assistance

**Reply by gbengaoluwadahunsi** [reply]
Let's try to convert the image file to either Digital Imaging and Communications in Medicine (DICOM) format or proprietary formats like Aperio (.svs) or Hamamatsu (.ndpi).

Let's see how this goes.

**Reply by kivuvarosekivuvan** [reply]
#### That works, thanks

![Screenshot from 2024-03-06 01-38-33](https://github.com/camicroscope/caMicroscope/assets/124133577/717b93c7-0148-4ec7-8632-763f02565955)

**Reply by kivuvarosekivuvan** [reply]
@gbengaoluwadahunsi Is that how the response should look like?

**Reply by gbengaoluwadahunsi** [reply]
The first column is showing an error -3001

**Comment by birm** [comment]
I'm going to drop this link here; http://camicroscope.org/docs/ 
In the quite likely case that something is unclear in the docs, we can work together to improve the documentation.

I'm going to be a little slow getting to all of the responses; so many interested people!

**Reply by kivuvarosekivuvan** [reply]
Thanks @birm, this will be helpful

**Reply by gbengaoluwadahunsi** [reply]
Thank you @birm

**Reply by nwanduka** [reply]
Thank you

**Reply by DIANAAJAMBO** [reply]
This is actually really helpful. Thank you

**Reply by mhdamaan79** [reply]
Thanks @birm

**Comment by NgweBecky** [comment]
@gbengaoluwadahunsi please can you help with a valid medical image. I keep getting errors
Thank you

**Reply by We13b-MD** [reply]
@Ngwe Becky @gbengaoluwadahunsi I am having issues setting rocker for my local machine for windows it keeps giving me an error saying virtual machine is not enabled

**Reply by gbengaoluwadahunsi** [reply]
Hi @We13b-MD, If I got you right, you seem to be trying to install docker on your machine.

what is the current free space on your machine, docker takes around  4-6gb of your CPU memory.
are you also installing via CMD or the GUI?
if the CMD appraoch is not wqrking, try install via the GUI.

**Reply by We13b-MD** [reply]
Thank you gbenga I tried installing it via CMD but had issues I will try via GUI and see how it goes

**Reply by NgweBecky** [reply]
@We13b-MD I started by downloading the docker desktop .exe file, use this link https://docs.docker.com/desktop/install/windows-install/ for guidance. Please reach out if any issue encountered in the process :)

**Reply by gbengaoluwadahunsi** [reply]
@We13b-MD , you are welcome, hope you have been able to fix that via the GUI?

**Reply by gbengaoluwadahunsi** [reply]
Hello @NgweBecky , @birm posted this link https://openslide.cs.cmu.edu/download/openslide-testdata/Aperio , you can get use any of the svs files, it worked and rocks.

**Reply by Khadijaarike** [reply]
@gbengaoluwadahunsi , this link is showing not found on my end, is there anything you did to access it? thanks for your response

**Reply by SisiVero** [reply]
Hello @We13b-MD , I faced the same issue.
First of you need to know the BIOS key of your laptop. How do you know that? By searching  "Your laptop model BIOS key". Once you get the answer,  restart your laptop and continue pressing the key, it will bring out the settings. Go to CPU configuration, scroll down to virtualization or virtual machine, and enable it. 
If you make use of HP Pavilion, I shared how I was able to solve mine here: https://github.com/orgs/camicroscope/discussions/690#discussion-6329605

Do let me know as it goes and if you need further assistant.

**Reply by We13b-MD** [reply]
ok thank you

**Reply by We13b-MD** [reply]
@SisiVero did the bios setup it worked thanks alot

**Reply by SisiVero** [reply]
@We13b-MD ,yourw welcome

**Comment by NgweBecky** [comment]
@birm @nanli-emory please any sugeestion on why I get a -3001 error when I upload a DICOM image.
After building the develop.yml file, I noticed the ca-load is not running and I am certain it leads to the slide upload issue. Please any solution or opinion?
![Screenshot (32)](https://github.com/camicroscope/caMicroscope/assets/54197600/04a5fc35-2cf4-47cd-a7c6-0bfe25c1a8ba)

**Reply by gbengaoluwadahunsi** [reply]
I think we need to install a dependency to fix ca-load or else we won't overcome that, with Mac users, that is easier to fix, I am trying now to figure out how to make ca-load work on my Windows, 

I might be migrating my development to the Mac/Ubuntu ecosystem henceforth, laughs.

**Reply by NgweBecky** [reply]
It is tough figure that out for me since the docs does not make mention. Thanks, I will have a look too

**Reply by gbengaoluwadahunsi** [reply]
Yes, that would be one of the suggestions we are going to make to the caMicroscope docs, the backend should be run through so that users shortly don't face the same glitch.

**Reply by nwanduka** [reply]
I'm experiencing the same issue that @NgweBecky described. Has anybody been able to overcome this?

**Reply by birm** [reply]
I think this may be related to this issue -- https://github.com/orgs/camicroscope/discussions/668
We've had some issues with loader at least relating to building/running on different platforms, as well as sometimes failing a required download. If you can get the logs for loader to determine why it exits, it can help narrow it down.

**Comment by Khadijaarike** [comment]
![Screenshot_7-3-2024_143435_localhost](https://github.com/camicroscope/caMicroscope/assets/161035744/48274dda-3ebd-41cc-bd19-e435182897ae)


has anybody been able to upload ? i will appreciate some help

**Reply by NgweBecky** [reply]
I have not @Khadijaarike. On my side, I noticed there is an issue on ca-load service which affects the slide upload task. Is the ca-load service running in container??

**Reply by Khadijaarike** [reply]
![Capture](https://github.com/camicroscope/caMicroscope/assets/161035744/75cad28b-ff9c-4ea7-b45e-b7b6aeb68148)


Thanks for your response, yes it is running..

**Reply by NgweBecky** [reply]
did you build the develop.yml or caMicroscop.yml file??

**Reply by gbengaoluwadahunsi** [reply]
@Khadijaarike, @birm posted a link for the demo images we can upload, try this https://openslide.cs.cmu.edu/download/openslide-testdata/Aperio/

**Reply by NgweBecky** [reply]
@gbengaoluwadahunsi  it shows not found for me, please help

**Reply by gbengaoluwadahunsi** [reply]
@NgweBecky  , click the link again

**Reply by NgweBecky** [reply]
works now, thanks

**Reply by gbengaoluwadahunsi** [reply]
@NgweBecky , you are welcome

**Reply by Khadijaarike** [reply]
@gbengaoluwadahunsi , THANK YOU so much ...though after uploading, its still showing not compatible

**Comment by gbengaoluwadahunsi** [comment]
did you download the svs image type pls?

**Reply by Khadijaarike** [reply]
![Screenshot_7-3-2024_201315_localhost](https://github.com/camicroscope/caMicroscope/assets/161035744/5b1d8f19-ad80-4916-b501-0d2c89d93738)


yes i did downloaded the svs

**Reply by gbengaoluwadahunsi** [reply]
@Khadijaarike , you ought to fill in the Filename with CMU-1.svs and the slidename input name too.

**Comment by We13b-MD** [comment]
Thank you very much will check it out

**Comment by nwanduka** [comment]
Has anyone been able to successfully upload an image? It appears a lot of us are having a hard time doing that.

**Comment by DIANAAJAMBO** [comment]
I need assistance with docker. So i have successfully installed it and also cloned the distro repo. I am stuck at running caMicroscope locally. I was going through the docker documentation and the project folder is supposed to have a Dockerfile but i am not able to locate it.

**Reply by gbengaoluwadahunsi** [reply]
hello @DIANAAJAMBO 
pls note the following,
1. Before you run your caMicroscope Distro repo in your local machine, make sure you have started the docker app in your machine.
2. Ensure that you afterwords go to your terminal, locate your folder and......
3 @birm run with docker-compose -f caMicroscope.yml up this will build all services and run in the foreground. Use docker-compose -f caMicroscope.yml build to rebuild the services. Once everything is up, go to http://localhost:4010/ to see the landing page.

Note: 6GB memory is recommended (in total, 2GB RAM + 4GB swap should work well)

**Reply by DIANAAJAMBO** [reply]
Thank you so much. Update is i did all that but it has been running for forever, Approximately how long does this take. @gbengaoluwadahunsi 
It keeps running and exiting with codes. 
![Screenshot (337)](https://github.com/camicroscope/caMicroscope/assets/124655553/2670f5a7-8020-4d53-8e52-b5a5a59f77bd)
I have been at it for the whole night.

**Reply by gbengaoluwadahunsi** [reply]
@DIANAAJAMBO @birm 
can  you try and build first before running the app
1. Use docker-compose -f caMicroscope.yml build to rebuild
2.  docker-compose -f caMicroscope.yml up

**Reply by SisiVero** [reply]
Hello @DIANAAJAMBO , of you've run the " docker-compose -f caMicriscope.yml" , it's okay for it to take a long time to run.
Mine took took days to fully run. Don't panic.
Also, from your respond I figured you already run and @gbengaoluwadahunsi  is suggesting you build, since you're currently running, you should allow it to run fully before building to avoid issues.
As stated in the instructions, you are to run before you build.

**Reply by gbengaoluwadahunsi** [reply]
@SisiVero @birm 
permit me to slide this in, I will suggest @DIANAAJAMBO  terminate the session she has and start over,  after discovering running took much time, I figured out it won't be nice for those who want to clone the repo to test to wait that long. I terminated my session, recloned and then ..........

1.  docker-compose -f caMicroscope.yml build 
2. docker-compose -f caMicroscope.yml up

It does work well, faster loading time with no glitch

Note: this is like you running npm build before trying npm start dev.
This "docker-compose -f caMicroscope.yml up" is attempting to build and then run at the same time, so it's the reason it kept the long of a time.

**Reply by DIANAAJAMBO** [reply]
@gbengaoluwadahunsi @SisiVero @birm let me terminate the session and start afresh. Thank you all for the help. I will definately give an update. 
PS; I am more familiar with the front end and in the mean time I would like to contribute more on the frontend as i get familiar with Docker. Any advise on where i should start.

**Reply by gbengaoluwadahunsi** [reply]
@DIANAAJAMBO , fork and clone this https://github.com/camicroscope/caMicroscope and have a good time contributing.

**Reply by SisiVero** [reply]
Okay. Since it's experimented and worked fine.
 Thank you @gbengaoluwadahunsi 
@DIANAAJAMBO do keep us updated.

**Reply by nwanduka** [reply]
I faced the same challenge while setting up the project. It seemed the session was running for too long. But I noticed that I didn't have to wait for it to run to completion to see the landing page. So, @DIANAAJAMBO, when you try to run it again, check the "port" column in the docker app (I've added a screenshot). If the 4010:4010 port appears, click it. That's the landing page.

You'll notice in the screenshot that not all the containers are running. So in the command prompt, it will still appear to be running without end. Try this and let us know how it goes.

![image](https://github.com/camicroscope/caMicroscope/assets/122698422/8efaaeb2-8672-4337-9a19-9129a6d90aad)

**Reply by DIANAAJAMBO** [reply]
@nwanduka Thank you so much. I looked for that port and it was able to access the landing page. @gbengaoluwadahunsi and @SisiVero, yes my docker was able to build and run very fast.  I also just noticed i needed to find the 4010 port like @nwanduka has showed me. Thanks to you all.

**Reply by nwanduka** [reply]
Awesome. I'm glad we figured that out.

**Comment by NgweBecky** [comment]
@gbengaoluwadahunsi please guide me on how to run the project locally if it there is a way. I have tried doing so using node and npm but it is not working, am I missing something?

**Reply by birm** [reply]
You can't run with just npm. I'd start with the instructions at https://github.com/camicroscope/distro

**Reply by gbengaoluwadahunsi** [reply]
Thank you @birm , I guess she is talking bout the caMicroscope viewer repo, @NgweBecky , just just have to locate the index.html file in the cloned repo and open it , it will pop up in your browser, you can then proceed with the various  html, css and JS files you want to work with.

**Reply by NgweBecky** [reply]
thank you @ and @birm, I did that earlier and thought I missed it. thanks again

**Reply by ayush-0110** [reply]
Hii, once we make changes in the UI using camicroscope repo, how can we view those changes on the site?

**Comment by We13b-MD** [comment]
I also had this issue I suggest get the rocker engine running first then you go to the terminal and write the command docker-compose -f caMicroscope.yml up
It takes time just be patient

**Reply by NgweBecky** [reply]
recommended to run first the
> docker-compose -f caMicroscope.yml build
> docker-compose -f caMicroscope.yml up

**Comment by We13b-MD** [comment]
And you need to have a solid internet connection

**Comment by We13b-MD** [comment]
@gbengaoluwadahunsi @SisiVero  @birm when building i got this error: OSError: libopenslide.so.0: cannot open shared object file: No such file or directory how do i solve this

**Reply by birm** [reply]
I think this is related to https://github.com/orgs/camicroscope/discussions/668. Can you provide some more logs, and tell us what system you're using?

**Reply by We13b-MD** [reply]
@ birm I am using WSL for windows ubuntu: 
![Screenshot 2024-03-08 112440](https://github.com/camicroscope/caMicroscope/assets/122805291/05d17691-a83f-4d4b-a2a0-693a593e158f)

**Reply by gbengaoluwadahunsi** [reply]
@birm , I  was able to load the image file but after the completion, it kept showing  -3001 and the terminal shows it's partly becos of the ca-load not connecting. 
error: OSError: libopenslide.so.0:

some of us too are facing similar glitches as mine, what should I do.

**Comment by We13b-MD** [comment]
Thank you @birm @gbengaoluwadahunsi @SisiVero it is up and running in my own case i had to install openslide library to solve the error:libopenslide.so.0: cannot open shared object file: No such file or directory. if anyone is having this error install the openSlide library you can read their documentation here:
https://openslide.org

**Reply by gbengaoluwadahunsi** [reply]
@We13b-MD , if i got you right, now you can upload an image file and view it?

**Reply by We13b-MD** [reply]
Not really i thought it worked as regarded the error:liboopenslide .so.0 that was the issue i had running the docker containers. i still have this
![Screenshot 2024-03-09 204415](https://github.com/camicroscope/caMicroscope/assets/122805291/52c7a2c2-cd89-41ec-a3b7-d8144c690587) tried svs files also but still working on it when i figure it out i will let you know and if you have it figured out please let me know how you did it

**Reply by DIANAAJAMBO** [reply]
@We13b-MD , Have you been able to resolve that error because i have the same. I could use some help.
![Screenshot (338)](https://github.com/camicroscope/caMicroscope/assets/124655553/32e19776-cb32-4d18-8541-38d46535c8f6)

**Reply by We13b-MD** [reply]
The solution is here check this link https://github.com/orgs/camicroscope/discussions/727#discussion-6356513

**Reply by DIANAAJAMBO** [reply]
I have found the discussion where this was resolved. Let me try it out. No worries. Below is where the resolution can be found for whoever encounters this error. 
https://github.com/orgs/camicroscope/discussions/727

**Comment by We13b-MD** [comment]
@SisiVero @NgweBecky have you been able to upload an image an if so please tell me how you did it

**Reply by NgweBecky** [reply]
No I have not

**Comment by blessdamy** [comment]
Hello everyone, trust that we are good. I have a question. After successfully cloning the repo and installing docker, what next  please?

**Reply by nwanduka** [reply]
The next thing would be to run caMicrscope in Docker. Follow these steps:
1. Open a terminal or command prompt window and navigate to the directory where you cloned the caMicroscope repository.
2. Run the following command:
`docker-compose -f develop.yml up`
This command starts the caMicroscope services defined in the develop.yml file. Ensure you have the docker desktop app open before running this command. It may take some time for Docker to download and set up the required containers.
3. Run the command: `docker-compose -f develop.yml build`
Once the Docker containers are up and running, open a web browser and go to http://localhost:4010/ to see the landing page.
4. Try to upload a slide.

**Reply by blessdamy** [reply]
Thank you so much @nwanduka
It took me days to get docker to work on my system..Hoping all things work now

**Comment by nwanduka** [comment]
Let's hope for the best.

**Comment by blessdamy** [comment]
@nwanduka it takes a really long time to load those commands

**Reply by nwanduka** [reply]
Yes, it will take a long time to load. I faced the same challenge. Here's what you should do: 

https://github.com/orgs/camicroscope/discussions/685#discussioncomment-8720763

**Comment by Moliki-Salman** [comment]
Hi everyone, I am finding it hard to understand how to upload a slide , please I want to clarify, does slides means to upload an image or a a file? Then what is the required file type to upload. I am using Mac Os. Please kindly guide me becaus ei have read through he chat and i could not get answers to my questions and possible directions to the solutions. Thank you


<img width="1280" alt="Screenshot 2024-03-12 at 4 31 53 PM" src="https://github.com/camicroscope/caMicroscope/assets/118880523/1b0130e4-263d-40a1-b926-b9ecc8be6c1a">

**Reply by nwanduka** [reply]
Hi @Moliki-Salman, 
1. Yes, a slide is an image.
2. The appropriate file type is a .svs file. Here's where you can [download suitable images](https://openslide.cs.cmu.edu/download/openslide-testdata/Aperio/).
3. If you're having issues with uploading an image, there's been a previous discussion about how to tackle it.
[Link to discussion](https://github.com/orgs/camicroscope/discussions/685#discussioncomment-8706786)
[Link to resolution](https://github.com/orgs/camicroscope/discussions/727)

I hope these answer your questions. If you experience any more challenges, please do not hesitate to drop your questions. @birm or any contributor who has been able to overcome the challenge can guide you through it.

I'd advise you go through the discussion threads first. You'll find some of the questions you may have have already been answered.

**Reply by Moliki-Salman** [reply]
Thanks @nwanduka for the insights

**Reply by sangwan-smriti** [reply]
what to do after uploading slide? where are tasks list given for the project?

**Comment by SanjanaSogimatt** [comment]
Hey everyone, It's nice to see everyone working together I wanted to know how to contribute to the code. At present I have the project running locally. Are we supposed to raise an issue regarding the feature development or should we just do a pr? How should I proceed further with this?

**Comment by gbengaoluwadahunsi** [comment]
Hello,
If what you are working on to contribute is already captured in the
contribution ReadMe file, you can proceed to fix it and make a PR, if not
you have to create an issue(bug or feature).

I hope this helps.

**Reply by SanjanaSogimatt** [reply]
Is it possible if you can point me to contribute readme file? It would be great help

**Reply by nwanduka** [reply]
Here you go @SanjanaSogimatt 

Contributing Guide: https://camicroscope.org/contributing.html
Contributor Tasks: https://github.com/camicroscope/GSOC/blob/master/ContributorTasks.md

**Reply by SanjanaSogimatt** [reply]
Thanks, @nwanduka

**Comment by Moliki-Salman** [comment]
@SanjanaSogimatt  and @gbengaoluwadahunsi  please where you able to run caracal  repo locally? Please could you put me through how you were able to successfully run the project locally

**Comment by gbengaoluwadahunsi** [comment]
Let me try and open my github account

**Comment by gbengaoluwadahunsi** [comment]
Here is the link to a simplified guidehttps://
github.com/orgs/camicroscope/discussions/712#discussioncomment-8727836

I am not with my PC, that is the reason it took me so long.

**Comment by Moliki-Salman** [comment]
@gbengaoluwadahunsi , Thank you for your  prompt response. However, I have done all what you described in your previous post. I have already created a discussion on the exact problem i am experiencing in a separate channel , so I will drop it here. PLEASE KINDLY GUIDE ME. 

 I am trying to setup the caracal repository locally and run it. I have cloned all the three required repository (caracal, Distro and caMicroscope) in a parent directory as instructed in the caracal readme.md. However, I am unable to run bash ./setup_script inside the caracal repo. whenever i try to run it, it throws " [ database ] : name not passed, set to default value (127.0.0.1)
[ database ] : name not passed, set to default value (camic)
mongo could not be found on path. Please ensure that mongo is installed and is on PATH". I have installed mongo but i am still unable to run the script.
Please kindly guide me on what to do to solve this problem.

Also, i want to clarify. The readme.md of caracal repo states that: "First clone the [Caracal (backup-dev branch)], but I could not see a branch named backup-dev branch under Caracal. I only cloned Caracal.

I look forward to your guidance. Thank you

**Reply by SanjanaSogimatt** [reply]
Even I am getting the same thing

**Reply by SanjanaSogimatt** [reply]
@Moliki-Salman did you figure out how to do it?

**Comment by gbengaoluwadahunsi** [comment]
Oh, I quite understand your point,

Unfortunately, I am facing the same problem as regards running the three
repositories on one server locally.

I think we should call on @birm  to put us through.

Thanks

**Comment by gbengaoluwadahunsi** [comment]
I will try it one more time and let you know.

**Comment by SanjanaSogimatt** [comment]
Guys, can anyone see the loading page being rendered while navigating through other segments in the page? Because I can't

**Comment by gbengaoluwadahunsi** [comment]
For which repo?

Distro or caMicroscope

**Reply by SanjanaSogimatt** [reply]
caMicroscope

**Comment by gbengaoluwadahunsi** [comment]
![sa](https://github.com/camicroscope/caMicroscope/assets/77133593/39973a36-60de-4a3f-9ce7-149f7a52b48f)
pls ignore the dark mode icon, that is my own contribution

**Reply by SanjanaSogimatt** [reply]
Ya so when you try to navigate, it shows a blank white screen rather then the loading animation

**Reply by gbengaoluwadahunsi** [reply]
Nope, it works perffectly
![Screenshot 2024-03-14 145345](https://github.com/camicroscope/caMicroscope/assets/77133593/96f2b268-2df1-4a74-8d9d-f8d734cbf8b3)

**Reply by SanjanaSogimatt** [reply]
Okay, cool

**Reply by gbengaoluwadahunsi** [reply]
alright.

**Comment by gbengaoluwadahunsi** [comment]
Here https://github.com/orgs/camicroscope/discussions/686#discussion-6325057

**Comment by Moliki-Salman** [comment]
@gbengaoluwadahunsi , @SanjanaSogimatt , Does this you have not been able to make a contributions on the project  too? Because I have not been able to move a step further ever since I am unable to run the repos.  Please kindly clarify because I feel hanged OR  are there other ways we can contribute pending the time that @birm  would be able to guide us on how to fix the problem

**Reply by SanjanaSogimatt** [reply]
Yup same here, I am on this for the whole day but still can't figure out. But we can contribute to other repos

**Comment by gbengaoluwadahunsi** [comment]
If you want to make contributions to the frontend development of the
project, you can kindly clone the caMicroscope repo and effect your changes
there and make a PR.

**Comment by sanketmp** [comment]
Hi all, I currently have a project running locally. Can someone or one of the maintainers guide me on what to do next? Apologies if I am missing anything.

**Comment by gbengaoluwadahunsi** [comment]
I hope it's clear now?

If you can't figure it out the Distro frontend  end, move to the
caMicroscope.

**Comment by gbengaoluwadahunsi** [comment]
Hi, which of the repo do you have running locally?

If it's the Distro, you have to try to upload a slide image and play around
with the view.

If it's the caMicroscope, you have to check the contribution ReadMe file
and  try to fix the issue in the list or suggest a feature on the frontend
of the project.

If it's the carcal repo, it's the backend codes for the project, check the
contribution ReadMe file and try to implement your contributions.


You can then push your contributions to the caMicroscope project by making
a PR.

I hope this helps.

**Reply by sanketmp** [reply]
Disto. Now I got it
